### PR TITLE
Add DotNetHowToSynonyms for synonyms sample code

### PR DESCRIPTION
**Pull request by mhko**
DotNetHowToSynonyms is a clone of DotNetHowTo with code samples for synonyms. 
This doesn't build; we need the 4.x version of SDK and update the reference in packages.config.
@brjohnstmsft, @chaosrealm, @HeidiSteen, @Yahnoosh

### CODE REVIEW

File: DotNetHowToSynonyms/DotNetHowToSynonyms/App.config
Comment (bitsforthought) [inline DotNetHowToSynonyms/DotNetHowToSynonyms/App.config]: This file has many unnecessary settings. Please use a copy of the App.config file in the DotNetHowTo project.

File: DotNetHowToSynonyms/DotNetHowToSynonyms/DotNetHowToSynonyms.csproj
Comment (bitsforthought) [inline DotNetHowToSynonyms/DotNetHowToSynonyms/DotNetHowToSynonyms.csproj]: Although currently this project lives in its own solution, you should change its ProjectGuid just in case we decide to combine it in a single solution with DotNetHowTo later. Each .csproj file should have a unique ProjectGuid. You'll need to change the reference in the .sln file too.

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Properties/AssemblyInfo.cs
Comment (bitsforthought) [inline DotNetHowToSynonyms/DotNetHowToSynonyms/Properties/AssemblyInfo.cs]: You should change the assembly info too so it's not the same as in the DotNetHowTo project.

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs
Comment (bitsforthought) [inline DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs]: You could probably delete RunQueries since it doesn't help tell the story of synonyms.

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs
Comment (bitsforthought) [inline DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs]: I don't think you need the stuff in the `#else` for the synonyms tutorial.

**Review by bitsforthought — CHANGES_REQUESTED** [review]

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Properties/AssemblyInfo.cs
Comment (mhko) [reply DotNetHowToSynonyms/DotNetHowToSynonyms/Properties/AssemblyInfo.cs]: Thanks

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs
Comment (mhko) [reply DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs]: Thanks.

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs
Comment (mhko) [reply DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs]: Removed.

File: DotNetHowToSynonyms/DotNetHowToSynonyms/App.config
Comment (mhko) [reply DotNetHowToSynonyms/DotNetHowToSynonyms/App.config]: Removed. Thanks

---
In reply to: [111008211](https://github.com/Azure-Samples/search-dotnet-getting-started/pull/10#discussion_r111008211) [](ancestors = 111008211)

File: DotNetHowToSynonyms/DotNetHowToSynonyms/DotNetHowToSynonyms.csproj
Comment (mhko) [reply DotNetHowToSynonyms/DotNetHowToSynonyms/DotNetHowToSynonyms.csproj]: Thanks. Generated a GUID and updated the two files. 

---
In reply to: [111008524](https://github.com/Azure-Samples/search-dotnet-getting-started/pull/10#discussion_r111008524) [](ancestors = 111008524)

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs
Comment (mhko) [reply DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs]: True. Done

---
In reply to: [111010602](https://github.com/Azure-Samples/search-dotnet-getting-started/pull/10#discussion_r111010602) [](ancestors = 111010602)

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs
Comment (bitsforthought) [inline DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs]: We should always encourage good security practices -- in this case, the principle of least privilege. Please keep the query key and admin key separate.

**Review by bitsforthought — CHANGES_REQUESTED** [review]

File: DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs
Comment (mhko) [reply DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs]: Okay. We should update DotNetSample too. 

---
In reply to: [111047416](https://github.com/Azure-Samples/search-dotnet-getting-started/pull/10#discussion_r111047416) [](ancestors = 111047416)

**Review by bitsforthought — APPROVED** [review]

**Comment by bitsforthought** [comment]
@mhko Everything looks good. I'll wait until Friday to merge this.

**Review by bitsforthought — APPROVED** [review]

**Pull request merged**
